test/jtreg-ext/requires/VMProps.java

```diff
@@ -462,7 +462,28 @@ protected String vmCDSForCustomLoaders() {
      * @return true if this VM can write Java heap objects into the CDS archive
      */
     protected String vmCDSCanWriteArchivedJavaHeap() {
-        return "" + ("true".equals(vmCDS()) && WB.canWriteJavaHeapArchive());
+        return "" + ("true".equals(vmCDS()) && WB.canWriteJavaHeapArchive()
+                     && isCDSRuntimeOptionsCompatible());
+    }
+
+    /**
+     * @return true if the VM options specified via the "test.cds.runtime.options"
+     * property is compatible with writing Java heap objects into the CDS archive
+     */
+    protected boolean isCDSRuntimeOptionsCompatible() {
+        String jtropts = System.getProperty("test.cds.runtime.options");
+        if (jtropts == null)
+            return true;
+        String CCP_DISABLED = "-XX:-UseCompressedClassPointers";
+        String G1GC_ENABLED = "-XX:+UseG1GC";
+        for (String opt : jtropts.split(",")) {
+            if (opt.equals(CCP_DISABLED))
+                return false;
+            if (opt.startsWith(GC_PREFIX) && opt.endsWith(GC_SUFFIX) &&
+                !opt.equals(G1GC_ENABLED))
+                return false;
```

Comment:
Based on the bug report both of these conditions are needed to be enabled for there to be a problem, but you exclude based on either `-XX:-UseCompressedClassPointers"` or any non-G1 GC. ??

Reply:
I've checked again, without the fix, the test fails if running with either `-Dtest.cds.runtime.options=-XX:-UseCompressedClassPointers` or any non-G1GC such as `-Dtest.cds.runtime.options=-XX:+UseZGC`.